docs/use-in-web-environment.md

```diff
@@ -0,0 +1,408 @@
+?>
+<?php $this->beginPage() ?>
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8"/>
+    <title><?= Html::encode($this->getTitle()) ?></title>
+    <?php $this->head() ?>
+</head>
+<body>
+<?php $this->beginBody() ?>
+    <h1>Posts</h1>
+    <?php foreach ($posts as $post): ?>
+        <h2><?= Html::encode($post->getTitle()) ?></h2>
+        <p><?= Html::encode($post->getDescription()) ?></p>
+    <?php endforeach; ?>
+<?php $this->endBody() ?>
+</body>
+</html>
+<?php $this->endPage() ?>
+```
+
+In order for scripts and tags registered in other places of the application to be correctly displayed
+in the call locations, special methods are called in the example above:
+
+- `beginPage()` - This method should be called at the very beginning of the view template.
+- `beginPage()` - This method should be called at the very end of the view template.
+- `head()` - This method should be called within the `<head>` section of an HTML page. It generates a placeholder which
+  will be replaced with the registered head HTML code (e.g. link tags, meta tags) when a page finishes rendering.
+- `beginBody()` - This method should be called at the beginning of the `<body>` section. It generates a placeholder
+  which will be replaced by the registered HTML code (e.g. СSS, JavaScript) targeted at the body begin position.
+- `endBody()` - This method should be called at the end of the `<body>` section. It generates a placeholder which
+  will be replaced by the registered HTML code (e.g. JavaScript) targeted at the body end position.
+
+When registering [CSS](#registering-css), [JavaScript](#registering-javascript) and [link tags](#registering-link-tags),
+you must specify the position in which this tag will be rendered. Positions are implemented by public constants:
+
+- `POSITION_HEAD` - Corresponds to the `head()` method.
+- `POSITION_BEGIN` - Corresponds to the `beginBody()` method.
+- `POSITION_END` - Corresponds to the `endBody()` method.
+- `POSITION_READY` - Corresponds to the `endBody()` method. Located after `POSITION_END`.
+- `POSITION_LOAD` - Corresponds to the `endBody()` method. Located after `POSITION_READY`.
+
+Every Web page should have a title. You can set the title in this way:
+
+```php
+$view->setTitle('My page title');
+```
+
+Then in the view, make sure you have the following code in the `<head>` section:
+
+```php
+<title><?= \Yiisoft\Html\Html::encode($this->getTitle()) ?></title>
+```
+
+## Rendering
+
+In addition to the `render()` and `renderFile()` methods, two `renderAjax()` and `renderAjaxString()`
+methods have been added for the WEB environment to render AJAX requests.
+
+The `renderAjax()` method is similar to `render()` except that it will surround the view being rendered with
+the calls of `beginPage()`, `head()`, `beginBody()`, `endBody()` and `endPage()`. By doing so, the method is
+able to inject into the rendering result with `JS/CSS` scripts and files that are registered with the view.
```

Comment:
```suggestion
The `renderAjax()` method is similar to `render()` except that it will surround the view being rendered with
the calls of `beginPage()`, `head()`, `beginBody()`, `endBody()` and `endPage()`. By doing so, the method is
able to inject JavaScript, CSS, and files, registered in the view template, into the rendering result.
```